server_addon/houdini/client/ayon_houdini/plugins/publish/collect_frames.py

```diff
@@ -20,82 +19,44 @@ class CollectFrames(plugin.HoudiniInstancePlugin):
-        else:
-            self.log.warning("Using current frame: {}".format(hou.frame()))
-            output = output_parm.eval()
-
-        _, ext = lib.splitext(
-            output, allowed_multidot_extensions=[
-                ".ass.gz", ".bgeo.sc", ".bgeo.gz",
-                ".bgeo.lzma", ".bgeo.bz2"])
+        output = output_parm.evalAtFrame(start_frame)
         file_name = os.path.basename(output)
-        result = file_name
-
-        # Get the filename pattern match from the output
-        # path, so we can compute all frames that would
-        # come out from rendering the ROP node if there
-        # is a frame pattern in the name
-        pattern = r"\w+\.(\d+)" + re.escape(ext)
-        match = re.match(pattern, file_name)
-
-        if match and start_frame is not None:
-
-            # Check if frames are bigger than 1 (file collection)
-            # override the result
-            if end_frame - start_frame > 0:
-                result = self.create_file_list(
-                    match, int(start_frame), int(end_frame)
-                )
 
         # todo: `frames` currently conflicts with "explicit frames" for a
         #       for a custom frame list. So this should be refactored.
+
         instance.data.update({
-            "frames": result,
+            "frames": file_name,  # Set frames to the file name by default.
             "stagingDir": os.path.dirname(output)
         })
 
-    @staticmethod
-    def create_file_list(match, start_frame, end_frame):
-        """Collect files based on frame range and `regex.match`
-
-        Args:
-            match(re.match): match object
-            start_frame(int): start of the animation
-            end_frame(int): end of the animation
-
-        Returns:
-            list
-
-        """
-
-        # Get the padding length
-        frame = match.group(1)
-        padding = len(frame)
-
-        # Get the parts of the filename surrounding the frame number,
-        # so we can put our own frame numbers in.
-        span = match.span(1)
-        prefix = match.string[: span[0]]
-        suffix = match.string[span[1]:]
-
-        # Generate filenames for all frames
-        result = []
-        for i in range(start_frame, end_frame + 1):
-
-            # Format frame number by the padding amount
-            str_frame = "{number:0{width}d}".format(number=i, width=padding)
-
-            file_name = prefix + str_frame + suffix
-            result.append(file_name)
-
-        return result
+        # Skip unnecessary logic if start and end frames are equal.
+        if start_frame == end_frame:
+            return
+
+        # Create collection using frame pattern.
+        # e.g. 'pointcacheBgeoCache_AB010.1001.bgeo'
+        # will be <Collection "pointcacheBgeoCache_AB010.%d.bgeo [1001]">
+        frame_collection, _ = clique.assemble(
+            [file_name],
+            patterns=[clique.PATTERNS["frames"]],
+            minimum_items=1
+        )
```

Comment:
This was just for my own sanity. But just confirming here that this also works for multi-dot extensions:
```python
import clique

frame_collection, _ = clique.assemble(
    ["test.hello.1001.ass.gz"],
    patterns=[clique.PATTERNS["frames"]],
    minimum_items=1
)
print(frame_collection)
# [<Collection "test.hello.%d.ass.gz [1001]">]

frame_collection, _ = clique.assemble(
    ["test.1001.bgeo.bz2"],
    patterns=[clique.PATTERNS["frames"]],
    minimum_items=1
)
print(frame_collection)
[<Collection "test.%d.bgeo.bz2 [1001]">]


frame_collection, _ = clique.assemble(
    ["test.1001.hello.1001.bgeo.bz2"],
    patterns=[clique.PATTERNS["frames"]],
    minimum_items=1
)
print(frame_collection)
# [<Collection "test.1001.hello.%d.bgeo.bz2 [1001]">]
```

Reply:
thanks for confirming.
As far as I tested this PR, it worked pretty well. 

tbh, I wish if we could use it directly with `$F` or `$F4` tokens. but I couldn't write the proper regex pattern.